public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -196,3 +233,72 @@ export function getPanelLinksBehavior(panel: PanelModel) {
     panelLinksMenu.setState({ links });
   };
 }
+
+function createExtensionContext(panel: VizPanel, dashboard: DashboardScene): PluginExtensionPanelContext {
+  const timeRange = sceneGraph.getTimeRange(panel);
+  let queryRunner = panel.state.$data;
+  let targets: DataQuery[] = [];
+  const id = getPanelIdForVizPanel(panel);
+
+  if (queryRunner instanceof SceneDataTransformer) {
+    queryRunner = queryRunner.state.$data;
+  }
+
+  if (queryRunner instanceof SceneQueryRunner) {
+    targets = queryRunner.state.queries;
+  }
+
+  if (queryRunner instanceof ShareQueryDataProvider) {
+    targets = [queryRunner.state.query];
+  }
+
+  let scopedVars = {};
+
+  // Handle panel repeats scenario
+  if (panel.state.$variables && panel.state.$variables instanceof SceneVariableSet) {
+    panel.state.$variables.state.variables.forEach((variable) => {
+      if (variable instanceof LocalValueVariable) {
+        scopedVars = {
+          ...scopedVars,
+          [variable.state.name]: { value: variable.getValue(), text: variable.getValueText() },
+        };
+      }
+    });
+  }
+
+  // Handle row repeats scenario
+  if (
+    (panel.parent instanceof SceneGridItem || panel.parent instanceof PanelRepeaterGridItem) &&
+    panel.parent?.parent instanceof SceneGridRow
+  ) {
+    const row = panel.parent.parent;
+    if (row.state.$variables && row.state.$variables instanceof SceneVariableSet) {
+      row.state.$variables.state.variables.forEach((variable) => {
+        if (variable instanceof LocalValueVariable) {
+          scopedVars = {
+            ...scopedVars,
+            [variable.state.name]: { value: variable.getValue(), text: variable.getValueText() },
+          };
+        }
+      });
+    }
+  }
```

Comment:
I have very similar code (but slightly different) in this PR: 

https://github.com/grafana/grafana/pull/78718/files#diff-6840fca137ac8050545634907394a5f61e286e98cd0ae30fe6893afa33eed619R47 

I can try after this is merged to refactor this and see if I can share / extract something to a util function